### PR TITLE
[BEAM-6428] Improve select performance with codegen

### PR DESCRIPTION
**Pull request by reuvenlax**
The schema select functionality is currently very inefficient - it walks through the schema resolution on each element selected in order to figure out how to properly distribute the select. Select is used by the select transform, but it is also used by the Group, Join, and any conversion transforms.

This PR analyzes the select predicate (e.g. a.b[].c.*), and generates bytecode to perform the select. All the schema resolution code that previously was done per element is now done once in order to determine what bytecode to generate.

### CODE REVIEW

**Comment by reuvenlax** [comment]
run dataflow validatesrunner

**Comment by reuvenlax** [comment]
run sql postcommit

**Comment by reuvenlax** [comment]
run dataflow validatesrunner

**Comment by reuvenlax** [comment]
run sql postcommit

**Comment by reuvenlax** [comment]
run dataflow validatesrunner

**Comment by reuvenlax** [comment]
run sql postcommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java [hunk 1]
```diff
@@ -28,8 +28,8 @@
 import org.apache.beam.sdk.schemas.FieldAccessDescriptor;
 import org.apache.beam.sdk.schemas.Schema;
 import org.apache.beam.sdk.schemas.utils.RowSelector;
-import org.apache.beam.sdk.schemas.utils.SelectByteBuddyHelpers;
```
Comment (alexvanboxel) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java]: I was wondering if SelectByteBuddyHelpers is still used after this removal (and would be able to be deleted). But I don't find it (org.apache.beam.sdk.schemas.utils.SelectByteBuddyHelpers) in the current master.

Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java]: This PR introduces SelectByteBuddyHelpers.

Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java]: I think you are looking at an individual commit.

**Comment by reuvenlax** [comment]
Run Java PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java (hunk 1 above)
Comment (alexvanboxel) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java]: oeps my bad. You are right (early morning). Resoving, looking further.

**Pull request merged**